### PR TITLE
boards: arm: stm32f103_mini: Enable USB for stm32f103_mini board

### PR DESCRIPTION
**Pull request by LongerHV**
Update stm32f103_mini board dts to enable USB. Change clock freqency to 72MHz and source to HSE.

Signed-off-by: Michał Mieszczak <michal@mieszczak.com.pl>

### CODE REVIEW

**Review by MaureenHelm — APPROVED** [review]

File: boards/arm/stm32f103_mini/stm32f103_mini.dts [hunk 1]
```diff
@@ -33,18 +33,23 @@
 };
 
 &clk_hsi {
+	status = "disabled";
+};
```
Comment (erwango) [inline boards/arm/stm32f103_mini/stm32f103_mini.dts:37]: Default status for clk_hsi node is disabled (in dtsi file).
So there is no value keeping this node with disabled status, please remove it.

**Review by erwango — CHANGES_REQUESTED** [review]
One comment otherwise LGTM.

Non mandatory nit: split changes in 2 commits (USB and clock change) so we're able to revert one or the other in case an issue is detected.

**Comment by erwango** [comment]
Thanks for the change but then please squash the commits into one.
In current case, deleting hsi node is considered as a fix of the first commit.
This could be fixed in initial commit, no reason to add a commit for that.

Splitting commits makes sense to split various unrelated changes.

**Comment by cfriedt** [comment]
@LongerHV - could you please address the comments by @erwango ?

**Comment by LongerHV** [comment]
@cfriedt @erwango I think all suggestions are addressed now.

**Review by erwango — APPROVED** [review]

**Comment by cfriedt** [comment]
@LongerHV  - please check the commit messages are in compliance. Otherwise, looks good to be merged.

**Comment by cfriedt** [comment]
🙌

**Pull request merged**